pkgs/development/python-modules/allure-python-commons/default.nix

```
@@ -0,0 +1,49 @@
+{ lib
+, fetchPypi
+, buildPythonPackage
+, pythonOlder
+, attrs
+, pluggy
+, six
+, allure-python-commons-test
+, setuptools-scm
+, flake8
+}:
+
+buildPythonPackage rec {
+  pname = "allure-python-commons";
+  version = "2.9.45";
+
+  disabled = pythonOlder "3.4";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "17alymsivw8fs89j6phbqgrbprasw8kj72kxa5y8qpn3xa5d4f62";
+  };
+
+  buildInputs = [
+    setuptools-scm
+  ];
+
+  propagatedBuildInputs = [ attrs pluggy six allure-python-commons-test ];
+
+  checkInputs = [
+    flake8
+  ];
+
+  checkPhase = ''
+    python -m doctest ./src/utils.py
+    python -m doctest ./src/mapping.py
+
+    flake8 src/
+  '';
```

```suggestion
  checkPhase = ''
    ${python.interpreter} -m doctest ./src/utils.py
    ${python.interpreter} -m doctest ./src/mapping.py
  '';
```
Running lint tests is only prone to new errors when we update flake8 but does not ensure the code is functioning correct.